### PR TITLE
Drive the menu via the highline gem.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Also rename `bin/dotd` to `bin/dotd_old` and (then) `bin/dotd_menu` to `bin/dotd`. Motivated by the suggestion [here](https://github.com/code-dot-org/code-dot-org/pull/17259#discussion_r134871784).

TODO(asher): Delete `bin/dotd_old` after we are firmly entrenched with the new system.

image

### CODE REVIEW

**Comment by ashercodeorg** [comment]
@islemaster: Feel free to recluse yourself if you are still dealing with DOTD.

File: bin/dotd [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
 
-# This script attempts to automate and guide the Developer Of The Day through
-# our deploy processes. In particular, it includes the AM DTT, the PM DTT with
-# content scoops from staging and levelbuilder, the DTP, and the DTL.
+# This script attempts to automate and guide the Developer Of The Day (DOTD) through our deploy
+# processes. As much of this process is automated, the script is now a menu driven script rather
+# than a linear workflow driven script.
+
+# TODO(asher): Rename this to `bin/dotd` (removing the linear workflow driven script) when
+# appropriate.
```
Comment (islemaster) [inline bin/dotd]: Remove TODO, it's now complete.

**Comment by islemaster** [comment]
Huh:
```
/home/brad/.rvm/gems/ruby-2.2.3/gems/bundler-1.14.6/lib/bundler/runtime.rb:40:in `block in setup':
You have already activated highline 1.7.8, but your Gemfile requires highline 1.6.21.
Prepending `bundle exec` to your command may solve this. (Gem::LoadError)
```
I ran bundle install after checking out the branch.  Am I missing a step?

File: bin/dotd (hunk 1 above)
Comment (ashercodeorg) [reply bin/dotd]: Done.

**Comment by ashercodeorg** [comment]
Re `dotd_menu_highline`: My fault, I had local changes that somehow failed to get committed. Should be fixed in d339558.

Re `highline 1.6.21`: For some reason, I ran into the same issues. Doing `gem uninstall highline` (choosing to uninstall all versions), then `bundle install`, allows me to run the script without prepending `bundle exec`. Not sure why this is necessary. That said, it isn't clear to me why `highline` is pre-existing in our `Gemfile` or `cookbooks/Gemfile.lock`, it doesn't appear in use elsewhere in our codebase.

**Comment by islemaster** [comment]
Well, I don't feel great about this:

image

Seems like we should sort out these dependency issues before we ship this change to the whole team.

I've gone ahead and done the manual uninstall and reinstall as you suggested, and when I run the script I get some additional unwanted output:

image

Looks like highline is generating its own menu with numbered options, and the single-letter keywords we've provided don't lend themselves to very nice formatting.

image

Seems like we're not using this library quite how it was  intended - let's provide nicer keywords for menu items and do less manual formatting of options ourselves.

**Comment by ashercodeorg** [comment]
Thanks for the pushback, in too much of a rush. Things should be cleaned up now. Sorry to have wasted your time.

**Pull request merged**